### DISCUSSION TITLE
Metamask locked issue (iOS)

### DISCUSSION THREAD

**Discussion by NasirAppelit** [WalletConnect/walletconnect-monorepo#3850] [Developer Support]
Hello all, 

I am using walletConnectswiftV2 to connect with external wallets like metamask. everything like connection and transaction is working but there is one issue.
connect work even if metamask is lock but all type of transaction doesnt show up on metamask if its locked. only when its unlocked does transaction approval popup shows.

what could be the problem?
how can i solve it?

after using :
Sign.instance.request(params: request)
i am opening metamask my self using this URL : 

"metamask://wc?uri=wc%3A54F16A88%2D53B1%2D41F3%2DAEED%2DCF0407554C09%401%3Fbridge%3Dhttps%253A%252F%252Fsafe%2Dwalletconnect%2Esafe%2Eglobal%26key%3Dbe73af0bfb0628c89e28c72ed19293d400fb67307ac91887130df643b8f54402"

**Comment by glitch-txs** [comment]
Hi, please open this issue in the Swift repository
